### DISCUSSION TITLE
Feature Request: Hotkey to increase/decrease volume by % amount set in settings

### DISCUSSION THREAD

**Discussion by datadragon00** [File-New-Project/EarTrumpet#677] [Ideas]
It would be good to have a user defined hotkey to increase or decrease the volume by a percentage amount you set in settings. For example ctrl-right arrow to increase volume by 5% or 10% at a time (as you set), or ctrl-left arrow to decrease volume by 5% or 10% at a time. Another one could be set to mute such as ctrl-down arrow puts the volume to 0% if you set to decrease by 100%. 
I think the hotkey should modify the volume of the currently used application rather then master. Currently you have to use the hotkey to open the panel, up or down arrow to the application being used, and then left and right arrows to modify.

**Comment by jupiter104** [comment]
I'd really like this as well. It would be great if you could assign hot keys for individual programs to increase and decrease their volumes.

**Reply by riverar** [reply]
@jupiter104, have you tried our latest [experimental dev build](https://eartrumpet.app/)? It comes with a new EarTrumpet Actions feature that lets you do just that.

**Comment by p1r473** [comment]
Would really love increase/decrease volume with a hotkey.

**Reply by riverar** [reply]
Hey @p1r473, have you tried our latest [experimental dev build](https://eartrumpet.app/)? It comes with a new EarTrumpet Actions feature that lets you do just that. Try it out!

**Reply by datadragon00** [reply]
Looks like good progress with the experimental, I dont yet see how to increase or decrease volume by a % such as 5% increase or decrease in the currently running app as I mentioned when hitting the hotkey you set. It seems to want me to set a fixed volume to change to but Im just trying to increase/decrease volume of the currently running app when you hit the hotkey by a certain percentage you set without having to open the app and then manually change it.  Please reply if you add it since I have to come to install the experimental rather than microsoft store update.

**Reply by p1r473** [reply]
> Hey @p1r473, have you tried our latest [experimental dev build](https://eartrumpet.app/)? It comes with a new EarTrumpet Actions feature that lets you do just that. Try it out!

Thanks, that worked great.

**Comment by ryanspain** [comment]
Many people try to avoid using the mouse where possible. EarTrumpet is the best tool for volume management and is very keyboard friendly but could be improved by adding support for greater volume adjustment steps as the OP as mentioned. 

I don't think a user setting is strictly needed here as 10% is a good step value considering smaller adjustments of 1% can be made as needed after some 10% adjustments. If configuration is desired, maybe it could be added in the future.

# Demo

**EDIT**: This feature is available in the [experimental dev build](https://github.com/File-New-Project/EarTrumpet?tab=readme-ov-file#experimental-dev-builds). See pull request [#1471](https://github.com/File-New-Project/EarTrumpet/pull/1471).

I've got a basic implementation of this done locally and I’ll create a PR shortly for it.

Illustrated below, the volume for a **Device** or **App** can be adjusted by 10% by holding `Ctrl` followed by pressing `Right` or `Left` to increase/decrease. Standard `Right`/`Left` inputs still work as normal to adjust volumes by 1%.

![Demo of increased volume stepping](https://github.com/File-New-Project/EarTrumpet/assets/8723433/aefe636a-93f7-4162-bae0-4dcdc05a5fcf)

## Why `Ctrl` + `Right`/`Left`?

The [World Wide Web Consortium](https://www.w3.org/WAI/ARIA/apg/patterns/slider/)(https://www.w3.org/WAI/ARIA/apg/patterns/slider/) suggest using `PageUp` and `PageDown` to increment/decrement slider values in larger steps (10% in this case), but I noticed that whether intentional or not, those keys move to the beginning/end of the list of apps. I opted not to change this.

When I initially tried to adjust the volumes using a greater step, I automatically used `Ctrl` with the`Right`/`Left` keys. It felt right to use this combination.
